src/goose/specs.clj

```diff
@@ -0,0 +1,117 @@
+(ns goose.specs
+  (:require
+    [goose.client :as c]
+    [goose.defaults :as d]
+    [goose.utils :as u]
+    [goose.worker :as w]
+
+    [clojure.spec.alpha :as s]
+    [clojure.spec.test.alpha :as st]
+    [clojure.string :as string]
+    [taoensso.nippy :as nippy]
+    [taoensso.carmine.connections :refer [IConnectionPool]]))
+
+; ========== Qualified Function Symbols ==============
+(defn- resolvable-fn? [func] (fn? @(resolve func)))
+(s/def ::fn-sym (s/and qualified-symbol? resolve resolvable-fn?))
+
+; ========== Redis ==============
+; Valid Redis URL patterns:
+; #1: redis://username:password@hostname:0-65353
+; #2: redis://hostname:0-65353
+(s/def :redis/url #(re-matches #"redis://.+:[0-9]{1,5}" %))
```

Comment:
When defining specs in a library, we should be careful to avoid potential name clashes. We should prefix our goose specs with `goose.`. We could rename this to something like `:goose.redis/url`, or `::redis-url` (which would expand to `:goose.specs/redis-url`. The `::` in front of a keyword will automatically prefix it with the current namespace. This is what `next.jdbc` has done. Obviously, we should rename all the other non-namespaced specs in this file as well.

Reply:
In this case, I think `goose.redis/url` would work better since the key name in the map is just `url`.

Reply:
Most keys are prefixed with `::`. 
`::url` won't do it because we want to differentiate between redis/rabbitmq. 
`::redis/url` doesn't work.
Most keys are `::broker-opts`, i.e. `:goose.specs/broker-opts`
Should I replace all `:redis/url` with `:goose.specs.redis/url` for consistency?

Reply:
@olttwa I think your last option i.e. `:goose.specs.redis/url` makes sense.